spatialdata/_core/models.py

```diff
@@ -0,0 +1,407 @@
+        coords
+            Coordinates of shapes.
+        shape_type
+            Type of shape.
+        shape_size
+            Size of shape.
+        transform
+            Transform of shape.
+        kwargs
+            Additional arguments for shapes.
+
+        Returns
+        -------
+        :class:`anndata.AnnData` formatted for shapes elements.
+        """
+
+        adata = AnnData(
+            None,
+            obs=pd.DataFrame(index=np.arange(coords.shape[0])),
+            var=pd.DataFrame(index=np.arange(1)),
+            **kwargs,
+        )
+        adata.obsm[cls.coords_key] = coords
+        if transform is None:
+            transform = Identity()
+        adata.uns[cls.transform_key] = transform
+        adata.uns[cls.attrs_key] = {"type": shape_type, "size": shape_size}
+        return adata
+
+
+class PointModel:
+    coords_key = "spatial"
+    transform_key = "transform"
+
+    def validate(self, data: AnnData) -> None:
+        if self.coords_key not in data.obsm:
+            raise ValueError(f"AnnData does not contain points coordinates in `adata.obsm['{self.coords_key}']`.")
+        if self.transform_key not in data.uns:
+            raise ValueError(f"AnnData does not contain `{self.transform_key}`.")
+
+    @classmethod
+    def parse(
+        cls,
+        coords: np.ndarray,  # type: ignore[type-arg]
+        var_names: Sequence[str],
+        transform: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> AnnData:
+
+        adata = AnnData(
+            None,
+            obs=pd.DataFrame(index=np.arange(coords.shape[0])),
+            var=pd.DataFrame(index=var_names),
+            **kwargs,
+        )
+        adata.obsm[cls.coords_key] = coords
+        if transform is None:
+            transform = Identity()
+        adata.uns[cls.transform_key] = transform
+        return adata
+
+
+class TableModel:
+    attrs_key = "spatialdata_attrs"
+
+    def validate(
+        self,
+        data: AnnData,
+        region: Optional[Union[str, Sequence[str]]] = None,
+        region_key: Optional[str] = None,
+        instance_key: Optional[str] = None,
+        **kwargs: Any,
+    ) -> AnnData:
+
+        attr = data.uns[self.attrs_key]
+        if "region" not in attr:
+            raise ValueError("`region` not found in `adata.uns['spatialdata_attr']`.")
+        if isinstance(attr["region"], list):
+            if "region_key" not in attr:
+                raise ValueError(
+                    "`region` is of type `list` but `region_key` not found in `adata.uns['spatialdata_attr']`."
+                )
+            if "instance_key" not in attr:
+                raise ValueError(
+                    "`region` is of type `list` but `instance_key` not found in `adata.uns['spatialdata_attr']`."
+                )
+
+        elif isinstance(attr["region"], str):
+            attr["region_key"] = None
+            attr["instance_key"] = None
+
+        return data
+
+    @classmethod
+    def parse(
```

Comment:
Some comments:
- `instance_key` is specified always, even when `region` is not a list
- no need to pass `region`, `region_key` and `instance_key` to the validator
I'll take care of modifying the above in the code that I will push.

Reply:
modified and pushed